### PR TITLE
cmd/protoc-gen-go-grpc: add change detector test

### PR DESCRIPTION
**Pull request by arvindbr8**
Fixes: https://github.com/grpc/grpc-go/issues/6748

RELEASE NOTES: none

### CODE REVIEW

File: .github/workflows/testing.yml
Comment (dfawley) [inline .github/workflows/testing.yml:100]: Nit: I think these can fit on one line.

File: .github/workflows/testing.yml
Comment (dfawley) [inline .github/workflows/testing.yml:130]: Nit: s/change detector// please.

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:16]: I don't believe this is strictly necessary since we never actually compile the pb.go files.

File: .github/workflows/testing.yml
Comment (dfawley) [inline .github/workflows/testing.yml:132]: Remove?

File: .github/workflows/testing.yml
Comment (dfawley) [inline .github/workflows/testing.yml:133]: This kind of thing should be in `protoc-gen-go-grpc_test.sh`.

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:18]: This will run the version of protoc-gen-go-grpc that's in your path?  I think you need to do:

```sh
go build .
PATH="$(pwd):${PATH}" protoc --go-grpc_out=. ......
```

OR find a way to run `protoc` where you point it at the binary directly instead of having it run out of your path using magic to parse flags.

File: regenerate.sh
Comment (dfawley) [inline regenerate.sh:32]: Please revert this file as discussed.

File: .github/workflows/testing.yml
Comment (arvindbr8) [reply .github/workflows/testing.yml]: I dont know of a clean one line way to apply `VET_SKIP_PROTO` to both `./vet.sh` commands

File: .github/workflows/testing.yml
Comment (arvindbr8) [reply .github/workflows/testing.yml]: Done

File: .github/workflows/testing.yml
Comment (arvindbr8) [reply .github/workflows/testing.yml]: Done

File: .github/workflows/testing.yml
Comment (arvindbr8) [reply .github/workflows/testing.yml]: Done

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (arvindbr8) [reply cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh]: Done

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (arvindbr8) [reply cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:17]: Done and tested: https://github.com/arvindbr8/grpc-go/actions/runs/8546553547/job/23417150415?pr=5

**Comment by arvindbr8** [comment]
The ugliness here ^^ is because i was trying to run the Github workflow

**Comment by arvindbr8** [comment]
`vet-proto` is expected to fail due to the `golden_grpc.pb.go` golden file that was added for the test.

File: .github/workflows/testing.yml
Comment (dfawley) [reply .github/workflows/testing.yml]: oh ... you need it set for both? :(

If you're interested, I've been meaning to delete `VET_SKIP_PROTO` and instead make it the default, with `VET_CHECK_PROTO=1` set if you actually what to check the proto files.

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:16]: Can you make this output to a `mktemp` please (and `go build` to it too)?  Then you don't need the `git clean`.  If you do the `git clean` then you run the risk of blowing away all uncommitted changes which means you need to confirm the repo is pristine first and ...ugh.

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:21]: Please wrap this and specify how to regenerate (`"run regenerate.sh"` right?)

**Comment by dfawley** [comment]
> vet-proto is expected to fail due to the golden_grpc.pb.go golden file that was added for the test.

That doesn't sound right, or are you saying it's because you were doing some temporary testing?  Please make sure vet-proto is passing.

**Review by arvindbr8 — COMMENTED** [review]
You are right. `vet-proto` is not supposed to fail. Fixed that and addressed your other comments.

File: .github/workflows/testing.yml
Comment (arvindbr8) [reply .github/workflows/testing.yml]: That's a good idea. I can send a follow for that.

Comment (arvindbr8) [reply .github/workflows/testing.yml]: https://github.com/grpc/grpc-go/pull/7099

-- I'll actually wait for that to merge before merging this PR

File: cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/protoc-gen-go-grpc_test.sh:36]: I think we can know the path, not search for it.

File: regenerate.sh
Comment (dfawley) [inline regenerate.sh:93]: Please try removing this again and see what is going on.

File: cmd/protoc-gen-go-grpc/proto/golden.proto
Comment (dfawley) [inline cmd/protoc-gen-go-grpc/proto/golden.proto:17]: google.golang.org/cmd/protoc-gen-go-grpc/proto.

But move this file and the output directory into a `testdata` directory.

**Review by dfawley — APPROVED** [review]

**Pull request merged**
